front/components/assistant_builder/actions/MCPToolsList.tsx

```diff
@@ -6,36 +8,46 @@ export function MCPToolsList({
   tools: { name: string; description: string }[];
 }) {
   return (
-    <>
-      <div className="heading-xl flex-grow pt-4 text-foreground dark:text-foreground-night">
+    <div>
+      <h4 className="heading-xl mb-3 flex-grow pt-4 text-foreground dark:text-foreground-night">
         Available tools
-      </div>
-
-      {tools ? (
-        <div className="space-y-0">
-          {tools.map(
-            (tool: { name: string; description: string }, index: number) => {
-              return (
-                <div
-                  key={index}
-                  className="flex flex-col gap-2 border-b border-border py-3 last:border-b-0 last:pb-0"
-                >
-                  <h4 className="heading-base flex-grow text-foreground dark:text-foreground-night">
-                    {asDisplayName(tool.name)}
-                  </h4>
-                  {tool.description && (
-                    <p className="copy-xs text-muted-foreground dark:text-muted-foreground-night">
-                      {tool.description}
-                    </p>
-                  )}
-                </div>
-              );
-            }
-          )}
-        </div>
+      </h4>
+      {tools.length !== 0 ? (
+        <p className="text-muted-foreground dark:text-muted-foreground-night">
+          No tools available
+        </p>
       ) : (
-        <p className="text-sm text-faint">No tools available</p>
+        <ContentMessage variant="primary" className="max-w-full">
```

Comment:
I'm not a big fan of [showing the dropdown](https://dust4ai.slack.com/archives/C050SM8NSPK/p1747126839916539?thread_ts=1747126446.937129&cid=C050SM8NSPK), what do you think of having a scrollable block instead? If you don't like it, I will give up and use the previous dropdown 😄